src/ng/directive/ngClass.js

```diff
@@ -97,15 +90,13 @@ function classDirective(name, selector) {
           oldModulo = newModulo;
         }
 
-        function ngClassOneTimeWatchAction(newClassValue) {
-          var newClassString = toClassString(newClassValue);
-
-          if (newClassString !== oldClassString) {
-            ngClassWatchAction(newClassString);
+        function ngClassWatchAction(newClassString) {
+          //When using a one-time binding the newClassString will return
+          //the pre-interceptor value until the one-time is complete
+          if (newClassString && typeof newClassString !== "string") {
```

Comment:
I've updated to `isString`. I kind of like an `if` block just so it's obvious that the comment applies to the full block, nothing more nothing less...